feat: add day01 plusfive solution

ezcomenezgo opened the pull request:
重新 git pull 下來衝突解完之後不知道為蝦咪一直落入圖片中的輪迴 QQ
後來直接把遠端的分支刪掉重新 push 一次

![image](https://user-images.githubusercontent.com/75515395/148487575-a53a3cac-1073-4ad4-a38e-826fd8f474e1.png)

謝謝圈圈的 review o(_ _)o

Commit:
feat. add day 1 JS Drum Kit plusfive's solution

Commit:
feat. add day 1 JS Drum Kit plusfive's solution

Commit:
fix. 修改 keyCode 改用 key + charCodeAt()

Commit:
fix. 修改 keyCode 改用 key + charCodeAt()

Commit:
修改 README

Commit:
修改 README

ezcomenezgo commented:
我想我可能稍微理解今天遇到的問題了！
我的問題跟哈奇今天列出來的不太一樣，我寫完 Day01 照著步驟 `fetch` + `rebase`，然後 `push` 了之後，發現大家在討論的 `keyCode` 已經不推薦使用的問題，所以我又回去我的 Day01 修改，修改完之後我又 `fetch` + `rebase` 了一次，這時發現 remote 的 `main` 在 local 的 `main` 前面，我又切去 local `main` 把最新進度 `pull` 下來，再切回去 Day01，為了保險起見又 `fetch` + `rebase` 了一次，然後在 `push` 時就出現了上面圖片中的錯誤。

那猜測原因是因為我當時的 local branch 跟 remote branch 的進度已經不一樣了，但確切不一樣的點我不太曉得（猜測可能是因為 main branch 有做變動，跟我第一次 `push` 上去保有的 `main` 的變動不同），不過總之就是因為不一樣，所以後來直接刪掉遠端的分支再 `push` 一次可能是比較保險的做法。

haha861924 committed:
feat: add day02 haha solution (#10)

* initialize

* feat: get key code and remove class

* style: adjust structure

* style: format

* style: 💄 adjust structure

* feat: 🎸 initialize

* feat: 🎸 add clock

* fix: 🐛 hour hand angle

* style: 💄 format

* refactor: 💡 day02 answer

* Create README.md

* style: 💄 remove ohther days files

Co-authored-by: haha861924 <qq861924@gmail.com>

wen-hsiu-hsu committed:
add day 2 solution - Kevin (#43)

* feat: 🎸 day 2 first commit

* feat: 🎸 add day 2 solution

* docs: ✏️ update kevin day 2 readme

* fix: 🐛 setInterval 觸發時間修改

* 我就亂推，day01 多東西了啦

* 一些我要保留的修改

* 於是我就直接刪掉 day01 多的東西

* Revert "於是我就直接刪掉 day01 多的東西"

This reverts commit 39147c296b1988251879bcb42c32f6619008f75d.

* Revert "一些我要保留的修改"

This reverts commit f496e4e32434c3f7f8b44744ebdf07c5f6b3b09f.

* Revert "我就亂推，day01 多東西了啦"

This reverts commit 6d98fec273129d4cb023e8c20a3fdbf43c9159a4.

lee926427 committed:
feat: 🎸 add day02 Recoil (#51)

wen-hsiu-hsu committed:
docs: ✏️ add Day 2 README.md

kayac-chang commented on day01/plusfive/main.js:
```diff
@@ -0,0 +1,28 @@
+// 監聽 document keydown 事件
+document.addEventListener('keydown', function(e) {
+  // 用抓到的 audio tag 跑迴圈
+  audiosArr.forEach(function(audio) {
```
可以用 `object` 或是 `Map` 來找對應的 audio
不用每次跑迴圈

ezcomenezgo replied:
好的！後來改用 Map 先把 audio 跟 block 都儲存起來
就不會在監聽的時候每次都跑迴圈了

kayac-chang commented on day01/plusfive/main.js:
```diff
@@ -0,0 +1,28 @@
+    let audioKey = Number(audio.dataset.key)
+    // 如果 event target 等於 迴圈跑到的 audio key 的話
+    if (e.key.toUpperCase().charCodeAt() === audioKey) {
+      // 就 play 它ㄅ
```
不用每段都上 comment
重點標示就好

ezcomenezgo replied:
收到！

kayac-chang commented on day01/plusfive/main.js:
```diff
@@ -0,0 +1,28 @@
+      // 就 play 它ㄅ
+      audio.play()
+      // 來加上 css 效果，用抓到的每個 div 跑迴圈
+      blocks.forEach(function(block) {
```
不用每次跑迴圈

kayac-chang commented on day01/plusfive/main.js:
```diff
@@ -0,0 +1,28 @@
+        // 如果 audio 迴圈跑到的 audio key 跟 div 的 key 一樣的話
+        if (Number(block.dataset.key) === audioKey) {
+          // 就加上 playing 的 class
+          block.classList.add('playing')
```
可以用 [classList.toggle](https://developer.mozilla.org/en-US/docs/Web/API/DOMTokenList/toggle) improve

ezcomenezgo replied:
已改～
謝謝哈囉

lee926427 committed:
refactor: 💡  day02 Recoil (#73)

* refactor: semantic code

* docs: refactor README.md

ericacadu committed:
Feature/erica day02 (#68)

* feat: add day02 Erica solution

* update: add readme & script comment

* refactor: Erica day02

wen-hsiu-hsu committed:
refactor: 💡 Kevin day 2 (#67)

kayac-chang committed:
Feat/day02 kim (#63)

* feat: ✨[Pedding]

* feat: 🎸 [Pedding] Clock can move by second

Clock can move but time need fix.

BREAKING CHANGE: 🧨 no

* Create js

* feat: 🎸 add day02 kim

add day02 kim solution.

* Delete js

* remove unuse folder

* fix: 🐛 Feat/day02 kim

Remove console,  add HTML file, fix README.

Co-authored-by: Kimberly <yantingfen@yantingfendeMacBook-Air.local>
Co-authored-by: KimberlyYen <87320011+KimberlyYen@users.noreply.github.com>

wen-hsiu-hsu committed:
docs: ✏️ update day 2 readme - add few deployed links

ezcomenezgo commented:
補充一下我遇到的狀況，在 Tailwind 翻譯上傳 git 步驟的文章裡面有提到～ 用 `git push -f` 就可以了
[請看「(5)將結果 push 至 git」的部分
](https://hackmd.io/0mc-X9-GRZK3rZ4DQ263KA#4-%E7%BF%BB%E8%AD%AF%E5%AE%8C%E6%88%90%E5%BE%8C-%E5%B0%87%E6%AA%94%E6%A1%88%E6%8E%A8%E9%80%81%E5%9B%9E-git)

HuskyHsu committed:
feat: 🎸 add day02 husky answer (#83)

* feat: 🎸 add day02 husky answer

* feat: 🎸 add transition timing function

ashramwen committed:
feat: 🎸 Add day02 cola solution (#87)

wen-hsiu-hsu committed:
docs: ✏️ update day 2 readme | add husky & cola page link

Commit:
改用 Map 的方式存取及加上監聽 keyup 事件

Commit:
修改 README

Commit:
Merge branch 'feature/day01_plusfive' of github.com:Rabbittee/JavaScript30 into feature/day01_plusfive

ashramwen commented:
神串留名
